internal/mode/static/handler.go

```diff
@@ -131,7 +121,47 @@ func (h *eventHandlerImpl) HandleEventBatch(ctx context.Context, logger logr.Log
 		}
 	}
 
-	h.cfg.statusUpdater.Update(ctx, buildGatewayAPIStatuses(graph, nginxReloadRes))
+	gwAddresses, err := status.GetGatewayAddresses(ctx, h.cfg.k8sClient, nil, h.cfg.gatewayPodConfig)
+	if err != nil {
+		logger.Error(err, "Setting GatewayStatusAddress to Pod IP Address")
+	}
+
+	h.cfg.statusUpdater.Update(ctx, buildGatewayAPIStatuses(graph, gwAddresses, nginxReloadRes))
+}
+
+func (h *eventHandlerImpl) handleEvent(ctx context.Context, logger logr.Logger, event interface{}) {
+	switch e := event.(type) {
+	case *events.UpsertEvent:
+		switch obj := e.Resource.(type) {
+		case *ngfAPI.NginxGateway:
+			h.updateControlPlaneAndSetStatus(ctx, logger, obj)
+		case *apiv1.Service:
+			podConfig := h.cfg.gatewayPodConfig
+			if obj.Name == podConfig.ServiceName && obj.Namespace == podConfig.Namespace {
```

Comment:
Seems like an extreme edge case that I don't think is worth us considering right now. We'd have to add logic to parse through all HTTPRoutes and their backends to see if this service matches...and then do something.

Reply:
what if we just handle it? So basically remove the else close?

In this case, it will be handled correctly. It might trigger unnecessary reload in rare case, but NGINX config will be correct.

```golang
			if obj.Name == podConfig.ServiceName && obj.Namespace == podConfig.Namespace {
				gwAddresses, err := getGatewayAddresses(ctx, h.cfg.k8sClient, obj, h.cfg.gatewayPodConfig)
				if err != nil {
					logger.Error(err, "Setting GatewayStatusAddress to Pod IP Address")
				}
				h.cfg.statusUpdater.UpdateAddresses(ctx, gwAddresses)
			
			h.cfg.processor.CaptureUpsertChange(e.Resource)
			
```

Reply:
I guess that works.